service/src/main/resources/db/changesets/20221011_unique_managed_application.yaml

```diff
@@ -0,0 +1,9 @@
+databaseChangeLog:
+  - changeSet:
+      id: unique_managed_application
+      author: mtalbott
+      changes:
+        - addUniqueConstraint:
+            constraint_name: unique_managed_application
+            tableName: billing_profile
+            columnNames: subscription_id, managed_resource_group_id
```

Comment:
Now I'm getting confused… an MRG can only be linked to a single billing profile. So why is it the combination of subscription_id and MRG that is unique? Wouldn't it be the combination of the primary key and `managed_resource_group_id`?

Reply:
The unique constraint will make sure there is only one record with a given subscription/MRG combo. Since a subscription/MRG combo represents a single managed app deployment, this constraint will trigger and throw an exception if we try to create a second record using the same subscription/MRG combo (managed app deployment)

Reply:
I was thinking that the `managed_resource_group_id` uniquely determined a MRG deployment… is that not the case?

Reply:
Hmm, I don't know. I thought that `managed_resource_group_id` was unique within a subscription (or maybe a tenant?), but that it wasn't necessarily globally unique. Azure's namespaces confuse me. If it is globally unique, then we can make just the `managed_resource_group_id` column unique.

Reply:
This link makes me think that the `managed_resource_group_id` isn't globally unique. I'm not sure if we're dealing with `deployment` or `resourcegroups`, but neither have a global naming scope